model-optimizer/extensions/middle/EltwiseInputReshape.py

```diff
@@ -65,58 +65,48 @@ def find_and_replace_pattern(self, graph: Graph):
+                # Create a map to know for which eltwise-consumers
+                # the same shape normalization is required
+                mapping = {}
+                output_port = node.out_port(out_port_idx)
+                output_port_shape = output_port.data.get_shape()
+                for consumer_port in output_port.get_destinations():
+                    if consumer_port.node.has_and_set('is_eltwise'):
+                        consumer_output_shape = consumer_port.node.out_port(0).data.get_shape()
+                        if np.array_equal(consumer_output_shape, output_port_shape):
+                            continue
+                        # Compute a new shape that is required for a shape normalization
+                        new_shape = output_port_shape
+                        if len(output_port_shape) != len(consumer_output_shape):
+                            for x in range(len(consumer_output_shape) - len(output_port_shape)):
+                                new_shape = np.insert(new_shape, 0, 1)
+                        new_shape = tuple([x for x in new_shape])
+                        if not new_shape in mapping:
+                            mapping.update({new_shape: [consumer_port]})
+                        else:
+                            mapping[new_shape].append(consumer_port)
 
-            if node.has_valid('value'):
-                # Check that requested shape are the same
-                # In case if they are different, we duplicate them
-                for shape_key in mapping.keys():
-                    shape = list(shape_key)
-                    new_value = np.reshape(node.value, shape)
-                    node_copy = Op.create_input_data_node(graph, node.id + '/copy', value=np.array(new_value))
-                    for consumer in mapping[shape_key]:
-                        edge_attrs = graph.get_edge_data(node.id, consumer.id)[0]
-                        del edge_attrs['new_shape']
-
-                        # Remove edge from previous data node and connect new data node with its consumer
-                        graph.remove_edge(node.id, consumer.id)
-                        graph.add_edge(node_copy.id, consumer.id, **edge_attrs)
-            else:
                 # Insert Reshape layer between data node and consumer
                 for shape_key in mapping.keys():
-                    shape = list(shape_key)
+                    new_shape = list(shape_key)
                     reshape_name = node.soft_get('name', node.id) + '/EltwiseReshape'
-                    reshape = Reshape(graph, attrs={'name': reshape_name})
-                    reshape_dim = Const(graph,
-                                        {'value': shape, 'name': reshape_name + '/Shape'}).create_node_with_data()
-                    reshape_data = reshape.create_node_with_data(inputs=[node, reshape_dim])
+                    reshape_node = create_op_with_const_inputs(graph, Reshape, {1: new_shape},
```

Comment:
Also since you modify the output shape of the operation you probably need the "override_output_shape" attribute and set `force_shape_inference = True` in the transformation

Reply:
Fixed.